core/src/main/java/org/elasticsearch/search/profile/SearchProfileShardResults.java

```diff
@@ -109,14 +106,21 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
 
     public static SearchProfileShardResults fromXContent(XContentParser parser) throws IOException {
         XContentParser.Token token = parser.currentToken();
-        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
         Map<String, ProfileShardResult> searchProfileResults = new HashMap<>();
-        ensureFieldName(parser, parser.nextToken(), SHARDS_FIELD);
-        ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.nextToken(), parser::getTokenLocation);
-        while((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
-            parseSearchProfileResultsEntry(parser, searchProfileResults);
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
```

Comment:
nit: we can check the expected token and then create the searchProfileResults map